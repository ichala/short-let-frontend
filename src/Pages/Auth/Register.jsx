import { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link, Navigate } from 'react-router-dom';
import { UserSignUp } from '../../Api/ApiCalls';

const Register = () => {
  const dispatch = useDispatch();
  const [formData, setFormData] = useState({});
  const [error, setError] = useState(null);
  const [Loading, setLoading] = useState(false);

  const handleChange = (e) => {
    setFormData({ ...formData, [e.target.id]: e.target.value });
    setError(null);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (formData.password !== formData.password_confirmation) {
      setError("password & password confirmation dosen't match ");
    } else {
      UserSignUp(formData, dispatch, setError, setLoading);
    }
  };
  const user = useSelector((state) => state.user);
  if (user) {
    return <Navigate to="/" replace />;
  }
  return (
    <>
      <div className="heading">
        <h1>Sign Up</h1>
        <hr />
      </div>
      {error && (
      <div className="alert alert-danger" role="alert">
        {error}
      </div>
      )}

      <div className="form-container">
        <form onSubmit={(e) => handleSubmit(e)}>
          <div className="form-group">
            First Name
            <input type="text" id="first_name" onChange={handleChange} className="form-control mb-2" required />

          </div>
          <div className="form-group">
            Last Name
            <input type="text" id="last_name" onChange={handleChange} className="form-control mb-2" required />

          </div>
          <div className="form-group">
            Email
            <input type="email" id="email" onChange={handleChange} className="form-control mb-2" required />

          </div>
          <div className="form-group">
            Password
            <input type="password" id="password" onChange={handleChange} className="form-control mb-2" required />

          </div>
          <div className="form-group">
            Password Confirmation
            <input type="password" id="password_confirmation" onChange={handleChange} className="form-control mb-2" required />
          </div>
          <div className="submit-btn mt-4">
            {!Loading ? (<button type="submit" className="btn btn-primary">Signup</button>) : (<div className="spinner-grow text-sm text-center text-primary" role="status" />)}
          </div>
        </form>
      </div>
      <div>
        <p className="mt-2">
          Already have an account?
          <Link className="ml-1" to="/login">Login</Link>
        </p>
      </div>

    </>
  );
};

export default Register;
