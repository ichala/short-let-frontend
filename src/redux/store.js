import { configureStore } from '@reduxjs/toolkit';
import authSlices from './slices/auth';

export default configureStore({
  reducer: {
    user: authSlices,
  },
});
