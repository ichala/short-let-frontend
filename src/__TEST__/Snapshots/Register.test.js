import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../redux/store';
import Register from '../../Pages/Auth/Register';

describe('Register', () => {
  it('should render the Register page', () => {
    const app = render(
      <Provider store={store}>
        <Router>
          <Register />
        </Router>
      </Provider>,
    );
    expect(app).toMatchSnapshot();
  });
});
