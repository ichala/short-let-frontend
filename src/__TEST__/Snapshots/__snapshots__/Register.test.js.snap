// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should render the Register page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="heading"
      >
        <h1>
          Sign Up
        </h1>
        <hr />
      </div>
      <div
        class="form-container"
      >
        <form>
          <div
            class="form-group"
          >
            First Name
            <input
              class="form-control mb-2"
              id="first_name"
              required=""
              type="text"
            />
          </div>
          <div
            class="form-group"
          >
            Last Name
            <input
              class="form-control mb-2"
              id="last_name"
              required=""
              type="text"
            />
          </div>
          <div
            class="form-group"
          >
            Email
            <input
              class="form-control mb-2"
              id="email"
              required=""
              type="email"
            />
          </div>
          <div
            class="form-group"
          >
            Password
            <input
              class="form-control mb-2"
              id="password"
              required=""
              type="password"
            />
          </div>
          <div
            class="form-group"
          >
            Password Confirmation
            <input
              class="form-control mb-2"
              id="password_confirmation"
              required=""
              type="password"
            />
          </div>
          <div
            class="submit-btn mt-4"
          >
            <button
              class="btn btn-primary"
              type="submit"
            >
              Signup
            </button>
          </div>
        </form>
      </div>
      <div>
        <p
          class="mt-2"
        >
          Already have an account?
          <a
            class="ml-1"
            href="/login"
          >
            Login
          </a>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="heading"
    >
      <h1>
        Sign Up
      </h1>
      <hr />
    </div>
    <div
      class="form-container"
    >
      <form>
        <div
          class="form-group"
        >
          First Name
          <input
            class="form-control mb-2"
            id="first_name"
            required=""
            type="text"
          />
        </div>
        <div
          class="form-group"
        >
          Last Name
          <input
            class="form-control mb-2"
            id="last_name"
            required=""
            type="text"
          />
        </div>
        <div
          class="form-group"
        >
          Email
          <input
            class="form-control mb-2"
            id="email"
            required=""
            type="email"
          />
        </div>
        <div
          class="form-group"
        >
          Password
          <input
            class="form-control mb-2"
            id="password"
            required=""
            type="password"
          />
        </div>
        <div
          class="form-group"
        >
          Password Confirmation
          <input
            class="form-control mb-2"
            id="password_confirmation"
            required=""
            type="password"
          />
        </div>
        <div
          class="submit-btn mt-4"
        >
          <button
            class="btn btn-primary"
            type="submit"
          >
            Signup
          </button>
        </div>
      </form>
    </div>
    <div>
      <p
        class="mt-2"
      >
        Already have an account?
        <a
          class="ml-1"
          href="/login"
        >
          Login
        </a>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
